### PR TITLE
Feature/fetch and use formio data

### PR DESCRIPTION
**Pull request by dpgalloway**

### CODE REVIEW

File: app/server/app/routes/api.js [hunk 1]
```diff
@@ -32,19 +33,47 @@ router.get("/bap", (req, res, next) => {
...
+    url += `/v/${req.query.version}`;
+  }
+  axios
+    .get(url, {
+      headers: {
+        "x-token": process.env.FORMIO_API_KEY,
+      },
+    })
+    .then((response) => {
+      res.json(response.data);
```
Comment (courtneymyers) [inline app/server/app/routes/api.js:49]: FYI, the forms schema/definition we need would be the entire JSON object returned in the response (there's no `data` key returned in this request – that's just for the submissions)

**Review by courtneymyers — COMMENTED** [review]
Thanks, Devin! This is great. Let's leave this branch open for right now, as I'm still using the mock data for building out the client app's UI. But this looks like the right approach for fetching the data.

We'll probably want to update the `env.example` file too so it's documented what environment variables are needed. We should also probably check that all needed environment variables are set too, as the app won't work without them – I'm doing that here: https://github.com/Eastern-Research-Group/csb-rebate-forms-app/blob/main/app/server/app/index.js#L18-L21 but we can maybe put all those checks in one place (either in the server's main file, or imported into it).

File: app/server/app/routes/api.js (hunk 1 above)
Comment (dpgalloway) [reply app/server/app/routes/api.js:49]: The response.data part is an axios thing - the axios response is an object that includes the request, config, the actual response data, and some other info (https://axios-http.com/docs/res_schema)

**Comment by dpgalloway** [comment]
> Thanks, Devin! This is great. Let's leave this branch open for right now, as I'm still using the mock data for building out the client app's UI. But this looks like the right approach for fetching the data.
> 
> We'll probably want to update the `env.example` file too so it's documented what environment variables are needed. We should also probably check that all needed environment variables are set too, as the app won't work without them – I'm doing that here: https://github.com/Eastern-Research-Group/csb-rebate-forms-app/blob/main/app/server/app/index.js#L18-L21 but we can maybe put all those checks in one place (either in the server's main file, or imported into it).

Thanks! I did update the .env.example file as part of this PR with the two form.io ones. Good call on checking that those environment variables exist on server startup. I'll make an update for that.

File: app/server/app/routes/api.js (hunk 1 above)
Comment (courtneymyers) [reply app/server/app/routes/api.js:49]: oh wow, thank you...shows how much I use axios. Coincidentally, a Form.io submission stores the submitted form fields data in a `data` object (along with other form metadata alongside it), so I assumed that's why you used that.

That said, the submissions returned are an array of objects too – so just ignore what I said, and plan to pass the response along as-is just like you're doing, and I can parse it as needed from the client app :smile:

**Comment by courtneymyers** [comment]
> Thanks! I did update the .env.example file as part of this PR with the two form.io ones.

Ha, sorry! I should've looked more carefully... thanks for doing that.

**Comment by courtneymyers** [comment]
@dpgalloway I haven't checked out your branch and installed `axios` and populated the formio API key yet, but conceptually, I think something like this would work for fetching the rebate form submissions:

```js
router.get("/rebate-form-submissions", ensureAuthenticated, (req, res) => {
  // TODO: pull UEIs from JWT – `ueis` array below is for demonstration purposes only
  const ueis = ["056143447853", "779442964145", "960885252143"];
  const query = ueis.join("&data.uei=");
  const url = `${process.env.FORMIO_BASE_URL}/submission?data.uei=${query}`;
  const data = axios
    .get(url, { headers: { "x-token": process.env.FORMIO_API_KEY } })
    .then((formioRes) => res.json(formioRes.data))
    .catch((formioErr) => {
      console.error(formioErr);
      const message = `There was an error retrieving Forms.gov submissions: ${formioErr.response.statusText}`;
      return res.status(formioErr.response.status || 500).json({ message });
    });

  res.json(data);
});
```

**Comment by courtneymyers** [comment]
@dpgalloway I know you were already working on that, and updating the formio request to limit what's being returned, so that URL would be modified... We can talk about what fields specifically would beed to be returned. At the very least the mongoDB ID, but any other fields that would ultimately be displayed in the user's dashboard table.

@coobr01 thoughts on what those specific fields would be?

**Comment by coobr01** [comment]
@courtneymyers If we are talking about the dashboard, UEI, Entity Name, Application Name, Last updated by, Last updated date. Note, I don't think all those fields are in GSA's v1 form def yet though.

**Comment by courtneymyers** [comment]
> @courtneymyers If we are talking about the dashboard, UEI, Entity Name, Application Name, Last updated by, Last updated date. Note, I don't think all those fields are in GSA's v1 form def yet though.

@coobr01 thanks, yes for the dashboard, and good to know. We'll just need to provide that feedback to the GSA folks once they have a new version of the form ready.

**Comment by courtneymyers** [comment]
@dpgalloway and @coobr01 – I can demo this when we meet this afternoon. It's ready to be merged in, but please feel free to take a look. You'll need to add the following environment variables, as shown in the `app/server/.env.example` file:
`FORMIO_PROJECT_URL`
`FORMIO_FORM_ID`
`FORMIO_API_KEY`

I'm using values from the "ERG CSB" project @coobr01 set up in the Forms.gov test project.

**Pull request merged**